charts/woodpecker/charts/agent/templates/statefulset.yaml

```diff
@@ -1,10 +1,11 @@
 apiVersion: apps/v1
-kind: Deployment
+kind: StatefulSet
```

Comment:
Maybe it is good idea to give user a choice between Deployment and Statefulset.

Reply:
Correct me if I am wrong but for the agent and current stateful approach with the agent config, `Deployments` are not that helpful. Also because they do not provide `volumeClaimTemplates` in their definition which is used/needed to provision a RWO per replica. They are most helpful when being used stateless or with a shared RWX volume.

In the end, multiple agents are mainly of interest for different node groups (e.g. different archs) as every agent just acts as a proxy which sends the pipeline specifications to standalone pods anyhow and doesn't run them inside the agent pod.

The server is more likely to profit from being Deployment, I agree there. Especially when SQLITE is not used and a PV is not required at all.

But for now, let's get the agent's fixed and finally stop the infinite registrations 😄

Reply:
You are right.
Some people just don't want Statefulset (me :). It messed my thoughts/comment a bit.
If / when Server becomes Deployment, I will wonder, why just little proxy is Statefulset still, while the main service is Deployment 😄 I don't remember if Server was Deployment ever, but in my env they are both Deployments. Therefore I was wondered/confused in 3023.

> But for now, let's get the agent's fixed and finally stop the infinite registrations

Agree. I have no more comments/suggestions.

Reply:
> why just little proxy (Agent) is Statefulset still, while the main service is Deployment 😄 

Server is fully stateless when being used with a non-sqlite DB. All configs are set via env vars and data is persisted in the DB.

Agents need to persist their `agent.conf` to avoid registering a new agent every time the pod is recreated.